pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -465,20 +468,19 @@ public void addExtraColumns(Schema newSchema) {
     _logger.info("Newly added columns: " + _newlyAddedColumnsFieldMap.toString());
   }
 
-  // NOTE: Okay for single-writer
-  @SuppressWarnings("NonAtomicOperationOnVolatileField")
   @Override
   public boolean index(GenericRow row, @Nullable RowMetadata rowMetadata)
       throws IOException {
     boolean canTakeMore;
+    int numDocsIndexed = _numDocsIndexed;
```

Comment:
snapshots `_numDocsIndexed` for a few reasons:

* This was actually safe, but code that generates warnings misleads people. It looks wrong and makes the reader ask questions. 
* On ARM (increasingly relevant hardware) reading a volatile variable incurs an actual LoadStore barrier.